src/socks5_stream.cpp

```diff
@@ -194,9 +194,11 @@ namespace libtorrent
 			m_buffer.resize(m_user.size() + m_password.size() + 3);
 			p = &m_buffer[0];
 			write_uint8(1, p);
-			write_uint8(m_user.size(), p);
+			TORRENT_ASSERT(m_user.size() < 0x100);
```

Comment:
where does 0x100 come from? I would expect the max length to be 255, or (std::numeric_limit<boost::uint8_t>::max())